### PR TITLE
CRIMAPP-626 Add PSE to workload report

### PR DESCRIPTION
**Pull request by timpeat**
## Description of change
Add separate rows for `initial` and `pse` in the workload report.
Show each workstream as a separate table in the workload report.
Store application_type on Review read model.

## Link to relevant ticket
[CRIMAPP-626](https://dsdmoj.atlassian.net/browse/CRIMAPP-626)

## Notes for reviewer
This is required for release of PSE

### Before changes:
Screenshot 2024-03-19 at 15 59 48

### After changes:
Screenshot 2024-03-19 at 15 59 09

## How to manually test the feature

[CRIMAPP-626]: https://dsdmoj.atlassian.net/browse/CRIMAPP-626?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: spec/read_models/received_on_reports/projection_spec.rb [hunk 1]
```diff
@@ -48,34 +50,50 @@
         travel_back
       end
 
-      let(:cat1_data) { dataset.fetch(cat1) }
-      let(:extradition_data) { dataset.fetch(extradition) }
+      let(:cat1_initial) { dataset.dig(cat1, initial) }
+      let(:cat1_pse) { dataset.dig(cat1, pse) }
+      let(:extradition_initial) { dataset.dig(extradition, initial) }
+      let(:extradition_pse) { dataset.dig(extradition, pse) }
```
Comment (hiboabd) [inline spec/read_models/received_on_reports/projection_spec.rb:56]: How come you have two extradition applications here when you only received one?

File: spec/system/reporting/snapshot_reports_spec.rb [hunk 2]
```diff
@@ -22,19 +22,6 @@
...
-      page.all('table thead tr.colgroup-details th').each_with_index do |el, i|
-        expect(el).to have_content expected_headers[i]
-      end
-    end
-
-    it 'shows the correct row headers for each work stream' do
-      column = all('table tbody tr th:first-child').map(&:text)
-      expect(column).to eq ['CAT 1', 'CAT 2', 'Extradition', 'Total']
-    end
-
```
Comment (hiboabd) [inline spec/system/reporting/snapshot_reports_spec.rb:37]: Why was this deleted?

File: spec/read_models/received_on_reports/projection_spec.rb (hunk 1 above)
Comment (timpeat) [reply spec/read_models/received_on_reports/projection_spec.rb:56]: These lets are for the values in the dataset. So, for example, extradition_pse is zero

File: spec/system/reporting/snapshot_reports_spec.rb (hunk 2 above)
Comment (timpeat) [reply spec/system/reporting/snapshot_reports_spec.rb:37]: It is tested in the workload_report spec and does not need to be tested here. This spec is for the general snapshot functionality (paging by time etc) and should not be testing the contents of the workload report.

File: spec/read_models/received_on_reports/projection_spec.rb (hunk 1 above)
Comment (timpeat) [reply spec/read_models/received_on_reports/projection_spec.rb]: I've appended _row to the lets to, hopefully, make it clearer.

**Review by hiboabd — APPROVED** [review]
Looks good!
